src/linter/block.go

```diff
@@ -1660,6 +1644,10 @@ func (b *BlockWalker) handleIf(s *stmt.If) bool {
 		b.ctx.sc.AddVarName(nm, types, "all branches", defCounts[nm] == linksCount)
 	}
 
+	// Collect all variable definitions from the if condition into the current context.
```

Comment:
Actually I don't think that we need to put _all_ assignments from the `if` condition into the current context. Also this code does not take into account that in a case like this:

```if($a = something()) { ... } else { ... }```

the `else` block would also have `$a` defined.

We only need to evaluate the first assignment in certain cases, for example:

```
if ( ($a = abc()) && ($b = bcd()) { ... } // $a is always defined, $b isn't
if ( ($a = abc()) || ($b = bcd()) { ... } // $a is always defined, $b isn't
if ($a = abc()) { ... } // $a is always defined
if (!$a = abc()) { ... } // // $a is always defined
```

And not in the other ones, for example in the following cases we don't want to handle any assignments:

```
if (abc() && ($b = bcd()) { ... } // $b might be defined
```